### ISSUE TITLE
The `whoami` command may end in StoreServerError [401] UNAUTHORIZED

### ISSUE THREAD

**Issue by facundobatista** [canonical/charmcraft#594]
This happens when the user is logged in (has local credentials) but those expired.

The solution is to wrap `Store.whoami` (in `store.py`) with the `_store_client_wrapper` decorator. But for this, the decorator should be enhanced.

So `_store_client_wrapper` should be a function that accepts a parameter and returns the real decorator.

All current decorator usage should mutate to `_store_client_wrapper()`, but in `whoami` it should be `_store_client_wrapper(auto_login=False)`.

Then, the code inside the decorator itself should be something like:

```
        except craft_store.errors.StoreServerError as error:
            if error.response.status_code == 401:
                if auto_login:
                    emit.progress("Existing credentials no longer valid. Trying to log in...")
                else:
                    raise NotLoggedIn()
            else:
                raise CommandError(str(error)) from error
```

**Comment by cmatsuoka** [comment]
401 error handing issue fixed in PR #609. CommandError conversion issue still open.

**Comment by sergiusens** [comment]
Fixed in #609 with the technical debt mentioned to be fixed once everything migrates to craft-cli
